x-pack/plugins/observability_solution/observability/public/pages/alert_details/hooks/use_add_investigation_item.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IHttpFetchError, ResponseErrorBody } from '@kbn/core/public';
+import {
+  CreateInvestigationItemParams,
+  CreateInvestigationItemResponse,
+} from '@kbn/investigation-shared';
+import { useMutation } from '@tanstack/react-query';
+import { useKibana } from '../../../utils/kibana_react';
+
+type ServerError = IHttpFetchError<ResponseErrorBody>;
+
+export function useAddInvestigationItem() {
```

Comment:
Do you think we can find a way to expose this hook from our plugin instead of duplicating them? Same for the create investigation? 
I don't know if that's possible or how we can do that. But we can try that later as well

Reply:
Definitely let's try to refactor this after this PR